src/common/Puppeteer.ts

```diff
@@ -23,6 +23,7 @@ import { ProductLauncher } from '../node/Launcher';
 import { BrowserFetcher, BrowserFetcherOptions } from '../node/BrowserFetcher';
 import { puppeteerErrors, PuppeteerErrors } from './Errors';
 import { ConnectionTransport } from './ConnectionTransport';
+import readPkgUp from 'read-pkg-up';
```

Comment:
Does this work in the browser as well as node?

Reply:
No. This is on my list to figure out. The old solution wouldn't have worked either.

I think it only needs to use the `package.json` in order to know which browser revision to download which is not needed in the browser; the work here is to try to prise that code out into a node specific function.